drivers/adodb-mysqli.inc.php

```diff
@@ -342,6 +342,12 @@ function GenID($seqname='adodbseq',$startID=1)
 
 		if ($rs) {
 			$this->genID = mysqli_insert_id($this->_connectionID);
+			if ($this->genID == 0)
+			{
+				$getnext = sprintf("select LAST_INSERT_ID() from %s;",$seqname);
+				$rs = @$this->Execute($getnext);
```

Comment:
What happens if execute() fails ?

Reply:
@anoopnaik you have not responded to my comment. 

Suppressing errors can cause problems, and while there may be valid reasons to do so, they need to be documented.